docs/tutorials/on-device-training.md

```diff
@@ -0,0 +1,1406 @@
+---
+title: On-Device Training
+nav_order: 11
+parent: Tutorials
+---
+
+# On-Device Training: Building an Android Application
+
+## Introduction
+
+In this tutorial, we will explore how to build an Android application that incorporates ONNX Runtime's On-Device Training solution. On-device training refers to the process of training a machine learning model directly on an edge device without relying on cloud services or external servers.
+
+We will guide you through the steps to create an Android app that can train a simple image classification model using on-device training techniques. This tutorial showcases the `transfer learning` technique where knowledge gained from training a model on one task is leveraged to improve the performance of a model on a different but related task. Instead of starting the learning process from scratch, transfer learning allows us to transfer the knowledge or features learned by a pre-trained model to a new task.
+
+For this tutorial, we will leverage the `MobileNetV2` model which has been trained on large-scale image datasets such as ImageNet (which has 1,000 classes). We will use this model for classifying custom data into one of four classes. The initial layers of MobileNetV2 serve as a feature extractor, capturing generic visual features applicable to various tasks, and only the final classifier layer will be trained for the task at hand.
+
+In this tutorial, we will use data to learn to:
+- Classify animals into one of four categories using a pre-packed animals dataset.
+- Classify celebrities into one of four categories using a custom celebrities dataset.
+
+## Prerequisites
+
+To follow this tutorial, you should have a basic understanding of Android app development using Java or Kotlin. Familiarity with C++ as well as familiarity with machine learning concepts such as neural networks and image classification will help as well.
+
+- Python development environment to prepare the training artifacts
+- Android Studio 4.1+
+- Android SDK 29+
+- Android NDK r21+
+- An Android device with a camera in [developer mode](https://developer.android.com/studio/debug/dev-options) with USB debugging enabled
+
+> **Note** The purpose of this tutorial is to show how ONNX Runtime can be used for On-Device Training. The author of this tutorial is not an expert in Android app development. The information provided in this tutorial is for educational purposes only and should not be considered as professional advice.
```

Comment:
This is too formal, not sure if we need this specially the last sentence. @kshama-msft @natke thoughts?

Reply:
Agree with Ashwini on this. Unless you think there is something that they definitely should not use in their own applications